extra_data/write_cxi.py

```diff
@@ -250,10 +301,174 @@ def write(self, filename, fillvalues=None):
                 )
                 mask.attrs['axes'] = axes_s
 
-            dgrp['experiment_identifier'] = h5py.SoftLink('/entry_1/experiment_identifier')
+            dgrp['experiment_identifier'] = h5py.SoftLink(
+                '/entry_1/experiment_identifier')
 
-            f['entry_1/data_1'] = h5py.SoftLink('/entry_1/instrument_1/detector_1')
+            f['entry_1/data_1'] = h5py.SoftLink(
+                '/entry_1/instrument_1/detector_1')
 
             dgrp.create_dataset('module_identifier', data=self.modulenos)
 
         log.info("Finished writing virtual CXI file")
+
+
+class XtdfCXIWriter(VirtualCXIWriterBase):
+    """
+    Machinery to write VDS files for a group of detectors with similar
+    data format - AGIPD, DSSC & LPD.
+
+    You don't normally need to use this class directly. Instead,
+    use the write_virtual_cxi() method on a multi-module detector
+    data interface object.
+
+    CXI specifies a particular layout of data in the HDF5 file format.
+    It is documented here:
+    http://www.cxidb.org/cxi.html
+
+    This code writes version 1.5 CXI files.
+
+    Parameters
+    ----------
+
+    detdata: extra_data.components.XtdfDetectorBase
+      The detector data interface for the data to gather in this file.
+    """
+    def __init__(self, detdata) -> None:
+        self._n_cells = 1
```

Comment:
This is a bit confusing, because these detectors all do have multiple memory cells. Maybe call the number something like 'cells per entry'? 'Entry' is the term I've used elsewhere for each part of the data along the first axis in the files.

Reply:
Thank you, renamed in https://github.com/European-XFEL/EXtra-data/commit/6582e9954511eb0bc752851d8f64a31bb7ed1348.